### PR TITLE
Add patient/therapist table and create new survey tables

### PR DESCRIPTION
**Pull request by aloomdar**

### CODE REVIEW

File: src/db/schema.ts [hunk 1]
```diff
@@ -25,18 +25,21 @@ export const createTable = pgTableCreator((name) => `betterhealth_${name}`);
 
 // Enums
 export const roleEnum = pgEnum("role", ["therapist", "member"]);
-export const surveyTypeEnum = pgEnum("survey_type", ["initial", "daily"]);
+export const questionnaireTypeEnum = pgEnum("questionnaire_type", [
+  "initial",
+  "daily",
+]);
```
Comment (jesuskcapellan) [inline src/db/schema.ts]: we dont need this anymore. the questionaire is just for signups

File: src/db/schema.ts [hunk 2]
```diff
@@ -50,19 +53,40 @@ export const feelingEnum = pgEnum("feeling", [
...
+export const specialtyEnum = pgEnum("specialty", [
+  "Anxiety",
+  "Depression",
+  "PTSD",
+  "Bipolar Disorder",
+  "Obsessive-Compulsive Disorder",
+  "Stress Management",
+  "Anger Management",
+]);
```
Comment (jesuskcapellan) [inline src/db/schema.ts]: These are just mental health disorders, we need specialties like addiction, family, health, etc. (find more dont just use these)

File: src/db/schema.ts [hunk 3]
```diff
@@ -50,19 +53,40 @@ export const feelingEnum = pgEnum("feeling", [
...
+export const foodHealthEnum = pgEnum("food_health", ["Low", "Medium", "High"]);
+export const sleepQualityEnum = pgEnum("sleep_quality", ["Poor", "Fair", "Good", "Excellent"]);
+export const selfImageEnum = pgEnum("self_image", ["Negative", "Neutral", "Positive"]);
+export const sleepTimeEnum = pgEnum("sleep_time", ["Before 8 PM", "8 PM - 10 PM", "10 PM - Midnight", "After Midnight"]);
```
Comment (jesuskcapellan) [inline src/db/schema.ts]: this should just be a number

File: src/db/schema.ts [hunk 4]
```diff
@@ -50,19 +53,40 @@ export const feelingEnum = pgEnum("feeling", [
...
+  "Depression",
+  "PTSD",
+  "Bipolar Disorder",
+  "Obsessive-Compulsive Disorder",
+  "Stress Management",
+  "Anger Management",
+]);
+export const foodHealthEnum = pgEnum("food_health", ["Low", "Medium", "High"]);
+export const sleepQualityEnum = pgEnum("sleep_quality", ["Poor", "Fair", "Good", "Excellent"]);
+export const selfImageEnum = pgEnum("self_image", ["Negative", "Neutral", "Positive"]);
```
Comment (jesuskcapellan) [inline src/db/schema.ts]: theses should just be integers from 1-5. the front end will handle the naming

File: src/db/schema.ts [hunk 5]
```diff
@@ -75,17 +99,59 @@ export const users = createTable(
...
 );
 
-// HealthHabits Table
-export const healthHabits = createTable("health_habits", {
+// Therapist Table with Authentication Fields
+export const therapist = createTable("therapist", {
+  therapistId: serial("user_id").primaryKey(),
+  email: varchar("email", { length: 255 })
+    .notNull()
+    .references(() => users.email),
```
Comment (jesuskcapellan) [inline src/db/schema.ts]: we dont need these in this table

File: src/db/schema.ts [hunk 6]
```diff
@@ -75,17 +99,59 @@ export const users = createTable(
...
+
+// Patient Table
+export const patient = createTable("patient", {
+  patientId: serial("user_id").primaryKey(),
+  agePreference: ageEnum("age").notNull(),
+  genderPreference: genderEnum("gender").notNull(),
+  specialtyPreference: specialtyEnum("specialty").notNull(),
+  email: varchar("email", { length: 255 })
+    .notNull()
+    .references(() => users.email),
```
Comment (jesuskcapellan) [inline src/db/schema.ts]: same here

File: src/db/schema.ts [hunk 7]
```diff
@@ -75,17 +99,59 @@ export const users = createTable(
...
+    .default(sql`CURRENT_TIMESTAMP`)
+    .notNull(),
+  updatedAt: timestamp("updated_at")
+    .default(sql`CURRENT_TIMESTAMP`)
+    .$onUpdate(() => sql`CURRENT_TIMESTAMP`),
+});
+
+// Daily Survey Table
+export const survey = createTable("survey", {
   habitId: serial("habit_id").primaryKey(),
```
Comment (jesuskcapellan) [inline src/db/schema.ts]: surveyid

File: src/db/schema.ts [hunk 8]
```diff
@@ -94,15 +160,58 @@ export const healthHabits = createTable("health_habits", {
...
+      .notNull(),
+    questionnaireData: json("questionnaire_data"),
+    createdAt: timestamp("created_at")
+      .default(sql`CURRENT_TIMESTAMP`)
+      .notNull(),
+    updatedAt: timestamp("updated_at")
+      .default(sql`CURRENT_TIMESTAMP`)
+      .$onUpdate(() => sql`CURRENT_TIMESTAMP`),
+  },
+);
```
Comment (jesuskcapellan) [inline src/db/schema.ts]: dont know why this was added? we just need a single questionaire table

File: src/db/schema.ts [hunk 9]
```diff
@@ -94,15 +160,58 @@ export const healthHabits = createTable("health_habits", {
...
-    .references(() => users.userId),
-  surveyDate: timestamp("survey_date").notNull(),
-  surveyType: surveyTypeEnum("survey_type").notNull(),
-  surveyData: json("survey_data"),
+    .references(() => therapist.therapistId),
+  questionnaireDate: timestamp("questionnaire_date").notNull(),
+  questionnaireData: json("questionnaire_data"),
   createdAt: timestamp("created_at")
     .default(sql`CURRENT_TIMESTAMP`)
     .notNull(),
```
Comment (jesuskcapellan) [inline src/db/schema.ts]: also not needed

File: src/db/schema.ts [hunk 10]
```diff
@@ -201,50 +310,50 @@ export const journals = createTable("journals", {
...
+//     .references(() => billing.billId),
+//     patientId: integer("patient_id")
+//     .notNull()
+//     .references(() => patient.patientId),
+//   changeType: changeTypeEnum("change_type").notNull(),
+//   changedAt: timestamp("changed_at")
+//     .default(sql`CURRENT_TIMESTAMP`)
+//     .notNull(),
+//   previousData: json("previous_data"),
+// });
```
Comment (jesuskcapellan) [inline src/db/schema.ts]: delete dont comment out

File: src/db/schema.ts [hunk 11]
```diff
@@ -25,18 +25,21 @@ export const createTable = pgTableCreator((name) => `betterhealth_${name}`);
...
+  "Approved",
+  "Declined",
 ]);
 export const appointmentStatusEnum = pgEnum("appointment_status", [
-  "pending",
-  "confirmed",
-  "cancelled",
+  "Pending",
+  "Confirmed",
+  "Cancelled",
```
Comment (jesuskcapellan) [inline src/db/schema.ts]: again why are we capitalizing it?

**Review by jesuskcapellan — CHANGES_REQUESTED** [review]

File: src/entities/surveys.ts
Comment (jesuskcapellan) [inline src/entities/surveys.ts:1]: you need to update this with the new values

File: src/data-access/surveys.ts
Comment (jesuskcapellan) [inline src/data-access/surveys.ts:1]: update these as well

**Review by jesuskcapellan — CHANGES_REQUESTED** [review]
Remember to do `npm run build && npm run start` to simulate the prod environment. You also need to do `npm run db:push` to push your changes to our remote database. I'm pretty sure this commit might break our db.

**Review by jesuskcapellan — APPROVED** [review]

**Review by tanvirlongia — APPROVED** [review]

**Pull request merged**